### PR TITLE
Created unit tests for PPC64 and PPC VLE disassembly

### PR DESCRIPTION
**Pull request by sprout42**
PPC64 disassembly unit tests currently fail, but PPC VLE disassembly pass with:
```
python -m unittest envi.tests.test_arch_ppc.PpcInstructionSet
```

### CODE REVIEW

**Comment by atlas0fd00m** [comment]
sorry sprout.  please carve out the changes for unit testing into a separate branch, based on my envi_ppc branch (without all the other commits).  this is too much to merge all at once.

File: envi/archs/ppc/const.py [hunk 1]
```diff
@@ -5,39 +5,48 @@
 
 TYPE_NONE = 0
 TYPE_REG =  1
-TYPE_IMM =  2
-TYPE_MEM =  3
-TYPE_JMP =  4
-TYPE_CR =   5
+TYPE_REG_SE = 2 # VLE specific register formats
```
Comment (atlas0fd00m) [inline envi/archs/ppc/const.py]: what are these VLE-specific register formats?

File: envi/archs/ppc/disasm_classes.py [hunk 2]
```diff
@@ -274,7 +274,7 @@ def render(self, mcanv, op, idx):
 
     def repr(self, op):
         val = self.getOperValue(op)
-        return '0x%x' % (val)
+        return '%#x' % (val)
```
Comment (atlas0fd00m) [inline envi/archs/ppc/disasm_classes.py]: damn!  i didn't know that!  learn something new every day.

File: envi/archs/ppc/vle.py [hunk 3]
```diff
@@ -81,15 +87,24 @@ def case_E_D8(types, data, va):
...
+    opers = ( op0(val0, val1, va), )
+    return opers
+
 def case_E_I16A(types, data, va):
     val1 = (data & 0x3E00000) >> 10;
     op1 = operands[types[0]]
     val0 = (data & 0x1F0000) >> 16;
     op0 = operands[types[1]]
     val1 |= (data & 0x7FF);
-    if (val1 & 0x8000):
```
Comment (atlas0fd00m) [inline envi/archs/ppc/vle.py]: no signedness in this one?

File: envi/archs/ppc/vle.py [hunk 4]
```diff
@@ -192,15 +240,26 @@ def case_E_BD24(types, data, va):
...
-    val0 = (data & 0xC0000) >> 18;
-    op0 = operands[types[0]]
     val1 = data & 0xFFFE;
     if (val1 & 0x8000):
         val1 |= 0xFFFF0000;
     
     op1 = operands[types[1]]
 
-    opers = ( op0(val0, va), op1(val1, va) )
+    if (data & 0x00200000) == 0x00200000:
```
Comment (atlas0fd00m) [inline envi/archs/ppc/vle.py]: does this simply want it's own decoder?

**Comment by sprout42** [comment]
ok, I'll split this into a PR for the unit test support and then make a different PR for the VLE disassembly fixes.

**Pull request closed without merging**